### PR TITLE
mgr/cephadm: remove_service: Feedback, if spec not found

### PR DESCRIPTION
**Pull request by sebastian-philipp**
`ceph orch rm my-unknown-service` still returned
`Removed service my-unknown-service`.

Signed-off-by: Sebastian Wagner <sebastian.wagner@suse.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: src/pybind/mgr/cephadm/module.py [hunk 1]
```diff
@@ -2016,9 +2018,13 @@ def remove_daemons(self, names):
...
-        self.spec_store.rm(service_name)
-        self._kick_serve_loop()
-        return ['Removed service %s' % service_name]
+        found = self.spec_store.rm(service_name)
+        if found:
+            self._kick_serve_loop()
+            return ['Removed service %s' % service_name]
+        else:
+            # must be idempotent: still a success.
+            return ['Failed to removed service %s: not found. ' % service_name]
```
Comment (jschmid1) [inline src/pybind/mgr/cephadm/module.py:2027]:
```suggestion
            return [f'Failed to remove service. <{service_name}> was not found.']
```

File: src/pybind/mgr/cephadm/tests/test_cephadm.py [hunk 2]
```diff
@@ -390,26 +390,6 @@ def test_remove_daemon(self, cephadm_module):
             out = wait(cephadm_module, c)
             assert out == ["Removed rgw.myrgw.myhost.myid from host 'test'"]
 
-    @mock.patch("cephadm.module.CephadmOrchestrator._run_cephadm", _run_cephadm(
```
Comment (jschmid1) [inline src/pybind/mgr/cephadm/tests/test_cephadm.py:393]: Care to re-add a unit test for this?

Comment (sebastian-philipp) [reply src/pybind/mgr/cephadm/tests/test_cephadm.py:393]: I did add this in https://github.com/ceph/ceph/pull/34617 but that would require some more cherry-picking. Sigh.

Comment (jschmid1) [reply src/pybind/mgr/cephadm/tests/test_cephadm.py:393]: ok, nvm

Comment (sebastian-philipp) [reply src/pybind/mgr/cephadm/tests/test_cephadm.py:393]: just added those commits to this PR.

**Comment by sebastian-philipp** [comment]
http://pulpito.ceph.com/swagner-2020-04-24_13:37:19-rados-wip-swagner2-testing-2020-04-24-1154-distro-basic-smithi/

Failure: https://tracker.ceph.com/issues/45194

**Pull request merged**